.github/workflows/release-trigger.yaml

```diff
@@ -0,0 +1,31 @@
+name: Check Version Update
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  check-version:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Push tag for each updated package
+        run: |
+          git config --global user.name "GitHub Action"
+          git config --global user.email "github-action@github.com"
+
+          for file in $(git diff --name-only HEAD~1..HEAD | grep pyproject.toml); do    
+              if git diff HEAD~1..HEAD $file | grep "version ="; then
+                  dirpath=$(dirname $file)        
+                  version=$(grep -o "version = .*" ${file} | cut -d '=' -f 2-)
+                  dirpath=$(dirname $file)
+                  relpath=${file#*/core|stackit/*}
+                  cleaned_version=$(echo "$version" | tr -d '" ')
+                  tag=$(echo "${dirpath}/${cleaned_version}")                
+                  git tag -a $tag -m "Release $version"        
+                  git push origin --tags          
```

Comment:
Also, does the pipeline have permissions to run this? Doesn't it need a token?

Reply:
The workflow permissions are set to "Read and write" so it should work without the need for a token.